packages/components/src/core/Autocomplete/components/AutocompleteGroup/index.tsx

```diff
@@ -0,0 +1,186 @@
+    FreeSolo
+  >;
+  onValueChange: (
+    colum: string,
+    event: React.SyntheticEvent,
+    value: AutocompleteValue<T, Multiple, DisableClearable, FreeSolo>,
+    reason: AutocompleteChangeReason,
+    details?: AutocompleteChangeDetails<T>
+  ) => void;
+  value?: AutocompleteValue<T, Multiple, DisableClearable, FreeSolo>;
+  popperOpen?: boolean;
+}
+
+type CustomAutocompleteProps<
+  T,
+  Multiple extends boolean | undefined,
+  DisableClearable extends boolean | undefined,
+  FreeSolo extends boolean | undefined
+> = Omit<
+  AutocompleteBaseProps<T, Multiple, DisableClearable, FreeSolo>,
+  | "renderInput"
+  | "options"
+  | "value"
+  | "onChange"
+  | "nonce"
+  | "rev"
+  | "rel"
+  | "autoFocus"
+  | "content"
+>;
+
+export type AutocompleteGroupProps<
+  T,
+  Multiple extends boolean | undefined,
+  DisableClearable extends boolean | undefined,
+  FreeSolo extends boolean | undefined
+> = CustomAutocompleteProps<T, Multiple, DisableClearable, FreeSolo> &
+  ExtraAutocompleteGroupProps<T, Multiple, DisableClearable, FreeSolo>;
+
+const AutocompleteGroup = <
+  T extends DefaultAutocompleteOption,
+  Multiple extends boolean | undefined,
+  DisableClearable extends boolean | undefined,
+  FreeSolo extends boolean | undefined
+>(
+  props: AutocompleteGroupProps<T, Multiple, DisableClearable, FreeSolo>
+) => {
+  const {
+    autocompleteProps,
+    onValueChange,
+    value: propValue,
+    multiple,
+    label,
+    InputBaseProps,
+    popperOpen,
+    inputValue,
+    PaperComponent,
+    ...rest
+  } = props;
+
+  const {
+    columnName,
+    columnWidth,
+    sdsIcon,
+    props: propProps,
+  } = autocompleteProps;
+
+  const [value, setValue] = useState<
+    AutocompleteValue<T, Multiple, DisableClearable, FreeSolo>
+  >(getInitialValue(multiple, propValue));
+  const [pendingValue, setPendingValue] = useState<
+    AutocompleteValue<T, Multiple, DisableClearable, FreeSolo>
+  >(getInitialValue(multiple, propValue));
+
+  useEffect(() => {
+    if (propValue !== undefined) {
+      setValue(propValue);
+    }
+  }, [propValue]);
+
+  const handleChange = useCallback(
+    (
+      event: React.SyntheticEvent,
+      newValue: AutocompleteValue<T, Multiple, DisableClearable, FreeSolo>,
+      reason: AutocompleteChangeReason,
+      details?: AutocompleteChangeDetails<T>
+    ) => {
+      propProps?.onChange?.(event, newValue, reason, details);
+      onValueChange(columnName, event, newValue, reason, details);
+
+      if (multiple) {
+        setPendingValue(newValue);
+      } else {
+        if (
+          newValue &&
+          !Array.isArray(newValue) &&
+          Object.prototype.hasOwnProperty.call(newValue, "name")
+        ) {
+          setValue(newValue);
+        }
+      }
+    },
+    [columnName, multiple, onValueChange, propProps]
+  );
+
+  return (
+    <StyledColumn
+      columnWidth={columnWidth}
+      className="SdsAutocompleteMultiColumn-column-root"
+    >
+      {sdsIcon && (
```

Comment:
hopefully not too much of a stretch, but should we also support ReactNode here, so users can construct their own Icon component and use it here?

Reply:
Ah, good catch! We're actually planning to enhance all components that accept SDSIcons to support React.Node, allowing users to utilize various SVG icons. I'll modify this section to ensure it can handle React.Node inputs. 😍

Reply:
This is actually way better :D thank you again @tihuan 

```javascript 
import Icon from "../core/Icon";

export const AUTOCOMPLETE_MULTI_COLUMN_OPTIONS = [
  {
    icon: <Icon sdsIcon="chevronRight" sdsSize="xs" sdsType="static" />,
    name: "Column One",
    options: [
    // ...items
    ],
    width: 260,
  },
  {
    icon: <Icon sdsIcon="chevronRight" sdsSize="xs" sdsType="static" />,
    name: "Column Two",
    options: [
    // ... items
    ],
    width: 260,
  },
  {
    name: "Column Three",
    options: [
    // ...items
    ],
    width: 180,
  },
];
```